ipykernel/inprocess/socket.py

```diff
@@ -31,7 +31,8 @@ def send_multipart(self, msg_parts, flags=0, copy=True, track=False):
     
     @classmethod
     def register(cls, other_cls):
-        warnings.warn("SocketABC is deprecated.", DeprecationWarning)
+        if other_cls is not DummySocket:
+            warnings.warn("SocketABC is deprecated.", DeprecationWarning)
```

Comment:
`stacklevel=2` ? it was not there before so I'm fine not changing that now.  Also since when is it deprecated (4.5.0 AFAICT).